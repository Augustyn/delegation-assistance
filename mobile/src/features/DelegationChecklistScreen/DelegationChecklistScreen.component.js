import React from "react";
import { View, Text, FlatList } from "react-native";

import ChecklistItem from "../../components/ChecklistItem";

import styles from "./DelegationChecklistScreenStyles.scss";

const renderChecklistItem = (checkbox, changeCheckboxState) => {
  return (
    <ChecklistItem
      data={checkbox.item}
      changeCheckboxState={changeCheckboxState}
    />
  );
};

const DelegationChecklistScreen = props => {
  const { title, checklist, changeCheckboxState } = props;

  return (
    <View>
      <Text style={[styles.title, styles.sideMargins]}>{title}</Text>
      <FlatList
        style={styles.list}
        data={checklist}
        renderItem={checkbox =>
          renderChecklistItem(checkbox, changeCheckboxState)
        }
      />
    </View>
  );
};

export default DelegationChecklistScreen;
