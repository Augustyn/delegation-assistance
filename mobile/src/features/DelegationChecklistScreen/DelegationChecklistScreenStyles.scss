@import "../../assets/styles/constants";

.title {
  text-align: center;
  font-size: 1.5rem;
  padding-bottom: 0.5rem;
  padding-top: 0.5rem;
  box-shadow: -1px 6px 5px;
  border-bottom-color: $inactiveTextColor;
  border-bottom-width: 1;
}

.sideMargins {
  margin-right: 20;
  margin-left: 20;
}

.list {
  margin-top: 5;
  padding-right: 20;
  padding-left: 20;
}
