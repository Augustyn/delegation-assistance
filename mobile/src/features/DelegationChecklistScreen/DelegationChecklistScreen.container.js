import React, { Component } from "react";
import { connect } from "react-redux";
import { fetchDelegationChecklist } from "../../actions/delegationChecklist.actions";
import {
  getDelegationChecklistName,
  getTasks
} from "../../selectors/delegationChecklist.selectors";
import DelegationChecklistScreen from "./DelegationChecklistScreen.component";

class DelegationChecklistScreenContainer extends Component {
  static navigationOptions = {
    title: "Checklist"
  };

  componentDidMount = () => {
    const delegationId = this.props.navigation.getParam("delegationId");
    this.props.fetchDelegationChecklist(delegationId);
  };

  changeCheckboxState = id => {
    //TODO: handle checkbox state change if necessary
  };

  _addKeysToItems = items => {
    //TODO: replace index with taskId when requirements will change
    return items.map((item, index) => {
      return Object.assign(item, { key: `${index}` });
    });
  };

  render() {
    return (
      <DelegationChecklistScreen
        title={this.props.checklistName}
        changeCheckboxState={this.changeCheckboxState}
        checklist={this._addKeysToItems(this.props.tasks)}
      />
    );
  }
}

const mapStateToProps = state => {
  return {
    checklistName: getDelegationChecklistName(state),
    tasks: getTasks(state)
  };
};

const mapDispatchToProps = {
  fetchDelegationChecklist
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DelegationChecklistScreenContainer);
