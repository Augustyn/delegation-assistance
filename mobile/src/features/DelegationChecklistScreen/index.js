import DelegationChecklistScreen from "./DelegationChecklistScreen.container";
export default DelegationChecklistScreen;
