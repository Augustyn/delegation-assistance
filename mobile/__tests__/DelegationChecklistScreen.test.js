import React from "react";
import { View } from "react-native";
import renderer from "react-test-renderer";
import { render, fireEvent, shallow } from "react-native-testing-library";

import DelegationChecklistScreen from "../src/features/DelegationChecklistScreen/DelegationChecklistScreen.component";

describe("DelegationChecklist snapshot", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<DelegationChecklistScreen />);
    expect(tree).toMatchSnapshot();
  });

  it("should render View", () => {
    const { getByType } = render(<DelegationChecklistScreen />);
    expect(getByType(View)).toBeDefined();
  });
});
