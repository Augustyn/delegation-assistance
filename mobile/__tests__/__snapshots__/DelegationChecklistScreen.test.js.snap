// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DelegationChecklist snapshot renders correctly 1`] = `
<View>
  <Text
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  />
</View>
`;
